engine/client/client.go

```diff
@@ -952,25 +997,70 @@ func allCacheConfigsFromEnv() (cacheImportConfigs []*controlapi.CacheOptionsEntr
 	return cacheImportConfigs, cacheExportConfigs, nil
 }
 
-type doerWithHeaders struct {
-	inner   graphql.Doer
-	headers http.Header
+func (c *Client) clientMetadata() engine.ClientMetadata {
+	return engine.ClientMetadata{
+		ClientID:                  c.ID,
+		ClientVersion:             engine.Version,
+		SessionID:                 c.SessionID,
+		ClientSecretToken:         c.SecretToken,
+		ClientHostname:            c.hostname,
+		UpstreamCacheImportConfig: c.upstreamCacheImportOptions,
+		UpstreamCacheExportConfig: c.upstreamCacheExportOptions,
+		Labels:                    c.labels,
+		CloudToken:                os.Getenv("DAGGER_CLOUD_TOKEN"),
+		DoNotTrack:                analytics.DoNotTrack(),
+	}
+}
+
+func (c *Client) AppendHTTPRequestHeaders(headers http.Header) http.Header {
+	headers = c.clientMetadata().AppendToHTTPHeaders(headers)
+	propagation.TraceContext{}.Inject(c.internalCtx, propagation.HeaderCarrier(headers))
```

Comment:
This should probably be `otel.GetTextMapPropagator()` so it'll propagate baggage as well (which is configured globally). Just recently started using baggage in #7605 for propagating things like `--debug` and `NO_COLOR=1` through to server-side `slog` usage, might build on it in the future for analytics.

Reply:
Fixed here https://github.com/dagger/dagger/pull/7315/commits/b8ab108ccc61b2ebb97b3558a15228ea0c03c861 thanks again!